### ISSUE TITLE
Pub documentation generator

### ISSUE THREAD

**Issue by DartBot** [dart-lang/pub-dartlang#294]
<a href="https://github.com/butlermatt"><img src="https://avatars.githubusercontent.com/u/1148886?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [butlermatt](https://github.com/butlermatt)**
_Originally opened as dart-lang/sdk#5130_

---

Per discussion here:
https://groups.google.com/a/dartlang.org/d/topic/misc/Ci89b9NX7h8/discussion

&quot;Additionally, what are the thoughts of pub generating dartdocs for a package either with a separate command (pub docs) or (pub install docs), or perhaps as part of the 'pub install' (with an extra flag?), generating and dartdoc documentation for any packages you have as dependencies, not just your own package.

I would definitely like to have pub support doc generation (the proper command for running dartdoc is a bit finicky).&quot;

To me there should be a two-part solution. First part is to generate dartdoc files for the package we own, that we are writing. Then another to be able to generate dartdoc files for packages we list as dependencies.

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

---

One option, we can fire an App Engine task when a package is uploaded, and a Dart VM running Compute Engine can pick up the task from App Engine's task queue, generate the docs, and post them Cloud Storage. It's so crazy, it just might work! :)

---

_Added this to the [Later](../milestones/Later) milestone._
_Added [Area-Pub](../labels/Area-Pub), [Triaged](../labels/Triaged) labels._

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

---

_Removed [Type-Defect](../labels/Type-Defect) label._
_Added [Type-Enhancement](../labels/Type-Enhancement) label._

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

---

_Removed this from the [Later](../milestones/Later) milestone._
_Added this to the [M3](../milestones/M3) milestone._

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

---

_Marked this as blocking dart-lang/sdk#7540._

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

---

_Marked this as blocking dart-lang/sdk#8072._

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

---

Bumping this out to M4 since the new doc UI is still in progress.

---

_Removed this from the [M3](../milestones/M3) milestone._
_Added this to the [M4](../milestones/M4) milestone._

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

---

Bumping again since the new doc UI is still ongoing.

---

_Removed this from the [M4](../milestones/M4) milestone._
_Added this to the [M5](../milestones/M5) milestone._

**Comment by DartBot** [comment]
<a href="https://github.com/amouravski"><img src="https://avatars.githubusercontent.com/u/264967?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [amouravski](https://github.com/amouravski)**

---

_Marked this as being blocked by dart-lang/sdk#8857._

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

---

I'm just going to mark this Milestone-Later for now until the new doc UI is further along.

---

_Removed this from the [M5](../milestones/M5) milestone._
_Added this to the [Later](../milestones/Later) milestone._

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

---

_Removed [Area-Pub](../labels/Area-Pub) label._
_Added [Area-PubSite](../labels/Area-PubSite) label._

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

---

Issue dart-lang/pub#820 has been merged into this issue.

**Comment by DartBot** [comment]
<a href="https://github.com/kasperl"><img src="https://avatars.githubusercontent.com/u/2156198?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kasperl](https://github.com/kasperl)**

---

_Removed this from the [Later](../milestones/Later) milestone._
_Added [Oldschool-Milestone-Later](../labels/Oldschool-Milestone-Later) label._

**Comment by DartBot** [comment]
<a href="https://github.com/kasperl"><img src="https://avatars.githubusercontent.com/u/2156198?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kasperl](https://github.com/kasperl)**

---

_Removed [Oldschool-Milestone-Later](../labels/Oldschool-Milestone-Later) label._

**Comment by DartBot** [comment]
<a href="https://github.com/nicolasgarnier"><img src="https://avatars.githubusercontent.com/u/3766663?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nicolasgarnier](https://github.com/nicolasgarnier)**

---

This issue has been moved to Github as part of a migration. Please use the Issue on Github going forward: https://github.com/dart-lang/pub-dartlang/issues

---

_Added [MovedToGithub](../labels/MovedToGithub) label._
